src/applyLookupTable.cc

```diff
@@ -0,0 +1,65 @@
+// -*- LSST-C++ -*- 
+
+/* 
+ * LSST Data Management System
+ * Copyright 2016 LSST Corporation.
+ * 
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the LSST License Statement and 
+ * the GNU General Public License along with this program.  If not, 
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#include <cstdint>
+#include <iostream>
+
+#include "lsst/pex/exceptions.h"
+#include "lsst/ip/isr/applyLookupTable.h"
+
+namespace lsst {
+namespace ip {
+namespace isr {
+
+template<typename PixelT>
+int applyLookupTable(
+    afw::image::Image<PixelT> &image,
+    ndarray::Array<PixelT, 1, 1> const &table,
+    PixelT indOffset
+) {
+    int numOutOfRange = 0;
+    auto const maxLookupCol = table.size() - 1;
+    for (auto col = 0, imHeight = image.getHeight(); col < imHeight; ++col) {
+        for (auto imPtr = image.row_begin(col), end = image.row_end(col); imPtr != end; ++imPtr) {
```

Comment:
My understanding is that the technique I used is significantly more efficient than linear iteration, as per this documentation from @RobertLuptonTheGood:

    // STL-compliant iterators are available.
    // However, they are not very efficient because the image data may not be contiguous
    // so these iterators must test for end-of-row on every increment.
    // (By the way, we do guarantee that an image's row data is contiguous).
    // iterator
    for (ImageT::iterator ptr = img.begin(), end = img.end(); ptr != end; ++ptr) {
        *ptr = 100;
    }

Unless that documentation is out of date I would rather not incur the cost.